### PR TITLE
pex 1.0

### PR DESCRIPTION
**Pull request by wickman**
This is just minor CHANGES.rst cleanup before minting 1.0.  But I'd like to take this opportunity to get feedback.  Is now the right time to cut 1.0?  Any features we're missing?  Don't like the semver contract?

### CODE REVIEW

**Comment by warsaw** [comment]
I've been playing a bit with master (not wickman/1.0 atm) and it looks pretty good w.r.t. use in creating Python snaps for Ubuntu core.  E.g.:

```
$ pex -o world.pex --python-shebang=/usr/bin/python3 --disable-cache -c world world
```

is a much nicer command than what I had been using.

One question comes to mind though: given that my setup.py for world has only one entry point:

```
    entry_points={
        'console_scripts': ['world = worldlib.__main__:main'],
        },
```
is it possible for pex to use that automatically if no -c option is given?  That doesn't seem to be the case currently.  It just means one less thing that *has* to be specified when building some snaps.  (IIRC, you were mentioning that pex could build multiple pex files if there were multiple console_scirpts defined.)

**Comment by wickman** [comment]
I don't think so.  First, you can specify any number of resolvables via the command line or via -r, so at that point things become ambiguous.  Second, not everyone wants to build a pex file with an entry point specified (instead they just want pex-as-virtualenv.)  Maybe it makes sense for this to be the default behavior of bdist_pex (see #99).

**Comment by warsaw** [comment]
Yeah, could be rather cool as bdist_pex.  Okay, fair enough!

File: CHANGES.rst [hunk 1]
```diff
@@ -2,9 +2,33 @@
...
+instead of requirement specs, ``pex -s`` has now been removed since source specs are accepted as arguments,
+and ``pex -p`` has been removed in favor of its alias ``pex -o``.
+
+The pex *command line interface* now adheres to semver insofar as backwards incompatible CLI
+changes will invoke a major version change.  Any backwards incompatible changes to the PEX
+environment variable semantics will also result in a major version change.  The pex *API* adheres
+to semver insofar as backwards incompatible API changes will invoke minor version changes.
+
+For users of the PEX API, it is recommended to add minor version ranges, e.g. ``pex>=1.0,<1.1``.
+For users of the PEX CLI, major version ranges such as ``pex>=1,<2`` should be sufficient.
```
Comment (Yasumoto) [inline CHANGES.rst]: +1

**Pull request merged**